Upstream-Status: Pending []

diff -Naur a/arch/arm64/boot/dts/freescale/imx943-orangebox-quantum.dts b/arch/arm64/boot/dts/freescale/imx943-orangebox-quantum.dts
--- a/arch/arm64/boot/dts/freescale/imx943-orangebox-quantum.dts	1970-01-01 01:00:00.000000000 +0100
+++ b/arch/arm64/boot/dts/freescale/imx943-orangebox-quantum.dts	2025-10-21 13:06:11.856807541 +0200
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2024-2025 NXP
+ */
+
+/dts-v1/;
+
+#include "imx943-ob2.dts"
+
+/ {
+	dummy_radio_codec: dummy_radio_codec {
+		#sound-dai-cells = <0>;
+		compatible = "linux,snd-soc-dummy";
+		// num-channels = <2>;
+    };
+
+    sound-radio {
+        compatible = "simple-audio-card";
+        simple-audio-card,name = "radio-dummy";
+        simple-audio-card,format = "i2s";
+        simple-audio-card,sound-dai = <&sai1>;
+        simple-audio-card,cpu {
+            sound-dai = <&sai1>;
+        };
+        simple-audio-card,codec {
+            sound-dai = <&dummy_radio_codec>;
+        };
+    };
+};
+
+&lpspi8 {
+	cs-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
+	num-cs = <1>;
+	pinctrl-0 = <&pinctrl_lpspi8>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	spidev1: spi@0 {
+		compatible = "rohm,dh2228fv";
+		reg = <0>;
+		spi-max-frequency = <12000000>;
+	};
+};
+
+&sai1 {
+	#sound-dai-cells = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai1>;
+	assigned-clocks = <&scmi_clk IMX94_CLK_AUDIOPLL1_VCO>,
+			<&scmi_clk IMX94_CLK_AUDIOPLL2_VCO>,
+			<&scmi_clk IMX94_CLK_AUDIOPLL1>,
+			<&scmi_clk IMX94_CLK_AUDIOPLL2>,
+			<&scmi_clk IMX94_CLK_SAI1>;
+	assigned-clock-parents = <0>, <0>, <0>, <0>,
+			<&scmi_clk IMX94_CLK_AUDIOPLL1>;
+	assigned-clock-rates = <3932160000>,
+			<3612672000>, <393216000>,
+			<361267200>, <24576000>;
+	fsl,sai-mclk-direction-output;
+	status = "okay";
+};
+
+&scmi_iomuxc {
+	
+	pinctrl_lpspi8: lpspi8grp {
+		fsl,pins = <
+			IMX94_PAD_GPIO_IO12__GPIO2_IO12			0x3fe
+			IMX94_PAD_GPIO_IO13__LPSPI8_SIN			0x3fe
+			IMX94_PAD_GPIO_IO14__LPSPI8_SOUT		0x3fe
+			IMX94_PAD_GPIO_IO15__LPSPI8_SCK			0x3fe
+		>;
+	};
+
+	pinctrl_sai1: sai1grp {
+		fsl,pins = <
+			IMX94_PAD_SAI1_RXD0__SAI1_MCLK				0x31e
+			IMX94_PAD_SAI1_TXC__SAI1_TX_BCLK			0x31e
+			IMX94_PAD_SAI1_TXFS__SAI1_TX_SYNC			0x31e
+			IMX94_PAD_SAI1_RXD0__SAI1_RX_DATA0			0x31e
+			IMX94_PAD_SAI1_TXD0__SAI1_TX_DATA0 			0x31e
+		>;
+	};
+
+};
diff -Naur a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
--- a/arch/arm64/boot/dts/freescale/Makefile	2025-10-21 13:04:24.744136510 +0200
+++ b/arch/arm64/boot/dts/freescale/Makefile	2025-10-21 13:09:29.074046627 +0200
@@ -611,6 +611,7 @@
 imx943-orangebox-mic-dtbs := imx943-orangebox.dtb imx943-orangebox-mic.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx943-orangebox-mic.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx943-orangebox-5G-modem.dtb
+dtb-$(CONFIG_ARCH_MXC) += imx943-orangebox-quantum.dts

 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx95-15x15-evk-flexspi-m2.dtb
diff -Naur a/sound/soc/soc-utils.c b/sound/soc/soc-utils.c
--- a/sound/soc/soc-utils.c	2025-10-21 12:52:13.619632365 +0200
+++ b/sound/soc/soc-utils.c	2025-10-21 13:11:55.910971411 +0200
@@ -21,6 +21,12 @@
 }
 EXPORT_SYMBOL_GPL(snd_soc_calc_frame_size);
 
+static const struct of_device_id soc_dummy_ids[] = {
+	{ .compatible = "linux,snd-soc-dummy", },
+	{ }
+};
+MODULE_DEVICE_TABLE(of, soc_dummy_ids);
+
 int snd_soc_params_to_frame_size(const struct snd_pcm_hw_params *params)
 {
 	int sample_size;
@@ -253,6 +259,7 @@
 static struct platform_driver soc_dummy_driver = {
 	.driver = {
 		.name = "snd-soc-dummy",
+		.of_match_table = of_match_ptr(soc_dummy_ids),
 	},
 	.probe = snd_soc_dummy_probe,
 };
